Update Tutorial for Okta - idp initiated (#1519)
FusionAuth/fusionauth-site: site/docs/v1/tech/identity-providers/samlv2-idp-initiated/okta.adoc

== Create and Partially Configure the Okta SSO Application

[NOTE]
====
You will want to set up this application as an admin. Okta allows users with various permissions. Later, create a user that can log into our newly created application, `Pied Piper`.
====

> Navigate to the Okta admin screen. Using the left "hamburger" menu, click on `Applications Header > Applications`. From here, click `Create App Integration`. Select the SAML option and hit next.

Reviewer: ```suggestion
Navigate to the Okta admin screen. Using the left "hamburger" menu, click on [breadcrumb]#Applications Header > Applications#. From here, click `Create App Integration`. Select the SAML option and hit next.
```

image::identity-providers/idp-initiated/okta-add-saml-service-provider-initial-screen.png[Setting up a SAMLv2 application in Okta,width=1200]

> Add an [field]#App name# describing the application. Configure any other "General Settings" as needed (including adding your award winning logo).

Reviewer: ```suggestion
Add an [field]#App name# describing the application. Configure any other [breadcrumb]#General Settings# as needed (including adding your award winning logo).
```

image::identity-providers/idp-initiated/okta-general-settings.png[General settings tab when creating a SAMLv2 application in Okta,width=1200,role=bottom-cropped]

> Click "Next". You will arrive at the "Configure SAML" section.

Reviewer: ```suggestion
Click "Next". You will arrive at the [breadcrumb]#Configure SAML# section.
```

image::identity-providers/idp-initiated/configure-saml.png[Settings tab when creating a SAMLv2 application in Okta,width=1200]

Please complete the [field]#Single sign on URL# and [field]#Audience URI (SP Entity ID)# with placeholder data. We will return to these fields in a moment (with real data from FusionAuth). Click next to proceed.

The final screen is related to marketing/support. This screen may or may not be present. Click the Finish button.

> image::identity-providers/idp-initiated/feedback-marketing-step.png[Feedback Marketing Step,width=1200]

Reviewer: ```suggestion
image::identity-providers/idp-initiated/feedback-marketing-step.png[Feedback Marketing Step,width=1200,role=bottom-cropped]
```

Now, under the Sign On tab, scroll down until you see a button called "View SAML setup instructions".  This will open a new window/tab. You will need the values displayed to complete the rest of this tutorial.

> image::identity-providers/idp-initiated/view-saml-setup-instructions[View SAML Setup Instructions,width=1200]

Reviewer: ```suggestion
image::identity-providers/idp-initiated/view-saml-setup-instructions.png[View SAML Setup Instructions,width=1200]
```

Reviewer: Please don't use arrows for docs. Please use red boxes as outlined in devdocsreadme.

```
Highlight sections using image preview editor
Highlights should be red rectangle with line weight 5
```

Record the [field]#Identity Provider Single Sign-On URL# and [field]#Identity Provider Issuer# values. The latter is a string such as `http://www.okta.com/exkq14ymac31Bx7895d6`.

> Download the "Identity Provider Certificate" too, then close the instructions tab.

Reviewer: ```suggestion
Download the "X.509 Certificate" too, then close the instructions tab.
```

> image::identity-providers/idp-initiated/detail-saml-setup-instructions[Details around certificate and SSO URL and issuer,width=1200]

Reviewer: ```suggestion
image::identity-providers/idp-initiated/detail-saml-setup-instructions.png[Details around certificate and SSO URL and issuer,width=1200]
```

Navigate to [breadcrumb]#Settings -> System -> CORS#.

> Determine the hostname and scheme of the Okta Identity Provider Single Sign-On URL. If the URL is `https://trial-6089629.okta.com/app/trial-6089629_test_1/exk1mn2r8kVgtAv6q697/sso/saml` then the hostname and scheme are `https://example.okta.com`. Additionally, you might have to add another entry with a trailing backslash `https://example.okta.com/`.

Reviewer: ```suggestion
Determine the hostname and scheme of the Okta Identity Provider Single Sign-On URL. If the URL is `https://trial-6089629.okta.com/app/trial-6089629_test_1/exk1mn2r8kVgtAv6q697/sso/saml` then the hostname and scheme are `https://trial-6089629.okta.com`. 
```

[NOTE]
====

> If you run into CORS configuration challenges, the best solution is to review the `System > Event Log`. This will tell you exactly what hostname and schemes require a whitelist.

Reviewer: ```suggestion
If you run into CORS configuration challenges, review the [breadcrumb]#System > Event Log# in the FusionAuth admin UI. This will tell you exactly what hostname and schemes must be added to the CORS allow lists.
```

====

> Add this value to the CORS [field]#Allowed origins# field. Ensure that the `POST` method is checked in the [field]#Allowed methods# field. Save the configuration

Reviewer: ```suggestion
Add this value to the CORS [field]#Allowed origins# field. Ensure that the `POST` method is checked in the [field]#Allowed methods# field. Save the configuration. 

Additionally, you might have to add another entry with a trailing backslash, such as `https://trial-6089629.okta.com/`.
```

== Complete SSO Configuration in Okta

Return to the Configure SAML Tab.  This can be found by navigating as follows from the App homepage: [breadcrumb]#General > SAML Settings > Edit Button > General Settings > Next Button > Configure SAML# tab in the Okta Admin screen.

> * Set the value of the [field]#Assertion Consumer Service URL# to the value of the [field]#Callback URL (ACS)# from the FusionAuth Identity Provider recorded above.

Reviewer: This looks like it needs to be updated.

> * Set the value of the [field]#Service Provider Entity Id# to the value of the [field]#Issuer# recorded above.

Reviewer: This looks like it needs to be updated.

> * Set the [field]#Application username format# to be `Email`.

Reviewer: This field name looks like it needs to be updated.

image::identity-providers/idp-initiated/okta-sign-on-advanced-settings.png[Configure Okta with the FusionAuth SP information,width=1200]

Save the Okta application by clicking "Next" until complete.

Navigate to the [breadcrumb]#General# tab and scroll to the [breadcrumb]#App Embed Link# section and note the [field]#Embed Link# value. This is the link a user needs to visit to begin the IdP initiated SSO, so you could place it in your application's navigation, launchpad, or elsewhere.

> Finally, click on the [breadcrumb]#Assignments# tab and assign the user to the application.

Reviewer: ```suggestion
Finally, click on the [breadcrumb]#Assignments# tab and assign the user to the application. First, select the user.
```

image::identity-providers/idp-initiated/okta-assign-application-to-user-step-one.png[Pick a user to assign to application step one,width=1200]

> 

Reviewer: ```suggestion

Then confirm the user should have access.

```

image::identity-providers/idp-initiated/okta-assign-application-to-user.png[Assigning a user to the SAML SP application step two,width=1200]

== Test It Out

> Open an incognito browser window and visit the [field]#Embed Link# value. This value can be found at the bottom of the General tab of your Okta Application. Log in with your Okta IdP credentials.

Reviewer: ```suggestion
Open an incognito browser window and visit the [field]#Embed Link# value. This value can be found at the bottom of the [breadcrumb]#General# tab of your Okta Application configuration, in the [breadcrumb]#App Embed Link# section. Log in with your Okta IdP credentials.
```
